op-deployer/pkg/deployer/state/state.go

```diff
@@ -118,5 +123,29 @@ type ChainState struct {
 
 	Allocs *GzipData[foundry.ForgeAllocs] `json:"allocs"`
 
-	StartBlock *types.Header `json:"startBlock"`
+	StartBlock *eth.BlockRef `json:"startBlock"`
+}
+
+func GetHeaderFromRpc(blockNumber *big.Int, l1Client *rpc.Client) (*eth.BlockRef, error) {
+	numberArg := ""
+	number := blockNumber
+	if number == nil {
```

Comment:
you can remove all this and just do:

```
var head eth.BlockRef
	err := l1Client.CallContext(ctx, &head, "eth_getHeaderByNumber", "latest")
if err != nil {
   return nil, fmt.Errorf("...")
  }
```

We don't need all this argument parsing logic. Let's keep it super simple for the time being until we have evidence that we'll need to update the header parsing elsewhere.